src/Diagnostics.DataProviders/KustoClient.cs

```diff
@@ -138,8 +138,20 @@ private DataTableResponseObjectCollection ProcessKustoResponse(string responseCo
             {
                 throw new Exception($"Kusto query response exceeded the Kusto record count limit: {_queryLimitDocsLink}");
             }
-
-            return JsonConvert.DeserializeObject<DataTableResponseObjectCollection>(responseContent);
+            try
+            {
+                return JsonConvert.DeserializeObject<DataTableResponseObjectCollection>(responseContent);
+            }
+            catch (Newtonsoft.Json.JsonSerializationException ex)
+            {
+                var responseObj = JsonConvert.DeserializeObject<DataTableExceptionResponseObject>(responseContent);
```

Comment:
Perhaps put this inside a try catch block ? This line could also possibly throw an exception just in case the returned response happens to be of some other format.